src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -150,8 +158,31 @@ public IndexService(Injector injector, Index index, @IndexSettings Settings inde
         indexFieldData.setIndexService(this);
         bitSetFilterCache.setIndexService(this);
         this.nodeEnv = nodeEnv;
+        clusterInfoService.addListener(new ShardSizeListener());
     }
 
+    /**
+     * Just records average size of shards in the cluster as a heuristic to help when allocating new shards to data paths.
+     */
+    class ShardSizeListener implements ClusterInfoService.Listener {
```

Comment:
I think for now we should just iterator over all teh shards on this node to get an idea....?